### PR TITLE
Fix detection of Piwik URL for overlay when using a custom piwik.php path for tracking

### LINKED ISSUE #9467: Function getPiwikUrlForOverlay returns wrong URL when it receives different URL than suffixed with piwik.php
If we are using e.g. "http://my.local/js/tracker.php" as piwikUrl in piwik.js, then Overlay tries to load following resource: https://my.local/js/tracker.phpplugins/Overlay/client/client.js?v=1 .

It's because of this line https://github.com/piwik/piwik/blob/2.15.0/js/piwik.js#L2632 that recognizes only URLs suffixed with "piwik.php".

Goal is to instead of removing previously defined suffix, create a function that will remove protocol://domain.tld[port]/ .

### PR DESCRIPTION
**Pull request by tsteur**
fixes #9467

### CODE REVIEW

**Comment by tsteur** [comment]
testcomment

**Comment by tsteur** [comment]
testcomment2

File: js/piwik.js [hunk 1]
```diff
@@ -2658,10 +2673,29 @@ if (typeof Piwik !== 'object') {
...
-                trackerUrl = trackerUrl.slice(0, trackerUrl.length - 9);
+            // if eg http://www.example.com/js/tracker.php?version=232323 => http://www.example.com/js/tracker.php
+            if (stringContains(trackerUrl, '?')) {
+                var posQuery = trackerUrl.indexOf('?');
+                trackerUrl   = trackerUrl.slice(0, posQuery);
+            }
+
+            if (stringEndsWith(trackerUrl, 'piwik.php')) {
+                // if eg without domain or path "piwik.php" => ''
+                trackerUrl = removeCharactersFromEndOfString(trackerUrl, 'piwik.php'.length);
```
Comment (kaz231) [inline js/piwik.js]: It's the same case like below (see else if section), so looks like it's unnecessary.

**Comment by sgiehl** [comment]
Besides the note from @kaz231  LGTM

File: js/piwik.js (hunk 1 above)
Comment (tsteur) [inline js/piwik.js]: I want to have this case there explicitly for a bit better readability and to make sure not to break anything compared to before as `piwik.php` is removed slightly differently below which should end in the same result but wasn't sure re side effects. I will remove it.

**Comment by tsteur** [comment]
@sgiehl I removed it

**Comment by sgiehl** [comment]
@tsteur Looks good to merge. would you resolve the conflicts and merge afterwards?

**Comment by tsteur** [comment]
I'll fix merge conflict and will merge afterwards

**Pull request merged**